linetools/lists/linelist.py

```diff
@@ -344,7 +349,195 @@ def all_transitions(self,line):
+                return self.__getitem__(name)
+            else:
+                return data
+
+    def available_transitions(self, wvlims, z, n_max=100,n_max_tuple=3, min_strength=4):
+        """For a given wavelength range in Angstroms, wvlims=(wv_min,wv_max),
+        this function retrieves the n_max_tuple strongest transitions per each ion 
+        species in the LineList available at redshift z within such a wavelength 
+        range. The output is sorted by strength of the strongest available transition 
+        per ion species, modualted by abundance.
+
+        Parameters:
+        ----------
+        wvlims : tuple of Quantity
+            Wavelength range, e.g. wvlims=(1100*u.AA, 3200*u.AA)
+        z : float
+            Redshift
+        n_max : int, optional
+            Maximum number of transitions retrieved
+        n_max_tuple : int, optional
+            Maximum number of transitions in a given ion species to 
+            retrieve. e.g., if Lyman series are all available, it will 
+            retrieve only up to Lyman gamma if n_max_tuple=3.
+        min_strength : float, optional
+            Minimum strenght calculated from log10(wrest * fosc * abundance)
+            In thin space HI 1215 has 14.7 [not]
+
+        
+        Returns:
+        ----------
+        dict (if only 1 transition found) or QTable (if > 1 transitions are found)
+        or None (if no transition is found)
+        """
+        if all(isinstance(n,int) for n in [n_max,n_max_tuple]):
+            pass
+        else:
+            raise SyntaxError('Both n_max and n_max_tuple must be integers!')
+
+        if n_max == 0:
+            return None
+
+        data = copy.deepcopy(self._data)
+
+        #identify unique ion_names (e.g. HI, CIV, CIII)
+        unique_ion_names = list(set([name.split(' ')[0] for name in data['name']]))
+        unique_ion_names = np.array(unique_ion_names)
+
+        #obtain the strongest transition of a given unique ion species
+        ion_name = []
+        strength = []
+        for ion in unique_ion_names: #This loop is necesary to have a non trivial but convinient order in the final output
+            aux = self.strongest_transitions(ion,wvlims,z,n_max=1) #only the strongest
+            if aux is not None:
+                if isinstance(aux,dict):#this should always be True given n_max=1
+                    name = aux['name']
+                else:
+                    name = aux['name'][0]
+                abundance = get_abundance(name)[0]
+                ion_name += [name]
+                strength += [np.log10(aux['wrest'].value * aux['f']) + abundance]
+        if len(ion_name)==0:
+            #no matches
+            return None
+
+        #create Table
+        unique = Table()
+        unique.add_column(Column(data=ion_name,name='name'))
+        unique.add_column(Column(data=strength,name='strength'))
+
+        #sort by strength
+        unique.sort(['strength'])
```

Comment:
Is not strictly needed but Table.sort() will convert the input to a list anyways in cases where input is not list already... so a if() statement is avoided.